src/java.base/share/classes/java/lang/Module.java

```diff
@@ -274,6 +274,12 @@ synchronized Module implAddEnableNativeAccess() {
     @PreviewFeature(feature=PreviewFeature.Feature.FOREIGN)
     public boolean isNativeAccessEnabled() {
         Module target = moduleForNativeAccess();
+        return isNativeAccessEnabled(target);
+    }
+
+    private static boolean isNativeAccessEnabled(Module target) {
+        if (target.enableNativeAccess)
```

Comment:
@minborg It'd seem a bit easier to implement using VarHandle and not having to use `synchronized`? Is the reason that you don't want to init VarHandle in this class? 🤔

Reply:
It is a bootstrap problem. We cannot use AtomicBoolean, VarHandle or Unsafe here.

Reply:
Shouldn’t `Unsafe` already be loaded at this point?